modules/heat_transfer/src/physics/HeatConductionPhysics.C

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HeatConductionPhysics.h"
+#include "ADHeatConduction.h"
+#include "HeatConductionTimeDerivative.h"
+
+InputParameters
+HeatConductionPhysics::validParams()
+{
+  InputParameters params = PhysicsBase::validParams();
+  params.addClassDescription("Add the heat conduction physics");
+
+  params.addParam<VariableName>("temperature_name", "T", "Variable name for the temperature");
```

Comment:
This will be better if we can have a `block` parameter with which the heat conduction physics can be block restricted. Checks will be need to make sure all boundary sides of the subdomain are covered with boundary conditions. We had this in Griffin for neutronics so that we can do multi-scheme neutron transport calculations.

Reply:
the block parameter exists, it's on the PhysicsBase class

I think those checks are valuable. They should have contributed to MOOSE. My vision for this would be a `[Diagnostics]` system that centralizes all these checks

Reply:
the Physics would pull in the diagnostics that are standard for each pde